src/openrct2-ui/UiContext.Linux.cpp

```diff
@@ -35,6 +35,8 @@ namespace OpenRCT2::Ui
     class LinuxContext final : public IPlatformUiContext
     {
     private:
+        mutable std::optional<bool> _hasFilePicker = std::nullopt;
```

Comment:
If no other methods are using this, you could make this a static variable in the HasFilePicker method instead.

Reply:
That would break a few principals.

Reply:
How so? What am I missing?

Reply:
Lets say you are writing a test for this class, you would get state shared between one test and the next.

Reply:
Right, and that wouldn't happen for properties. Gotcha. Thanks.